cmd/common/common.go

```diff
@@ -15,47 +17,50 @@ import (
 	"github.com/spf13/cobra"
 )
 
-var (
-	Profile *profiles.Profile
-
-	OtdfctlCfg config.Config
-)
+var OtdfctlCfg config.Config
 
 // InitProfile initializes the profile store and loads the profile specified in the flags
 // if onlyNew is set to true, a new profile will be created and returned
 // returns the profile and the current profile store
-func InitProfile(c *cli.Cli, onlyNew bool) (*profiles.Profile, *profiles.ProfileStore) {
+func InitProfile(c *cli.Cli) *profiles.OtdfctlProfileStore {
 	var err error
 	profileName := c.FlagHelper.GetOptionalString("profile")
 
-	Profile, err = profiles.New()
-	if err != nil || Profile == nil {
-		c.ExitWithError("Failed to initialize profile store", err)
+	hasKeyringStore, err := osprofiles.HasGlobalStore(config.AppName, osprofiles.WithKeyringStore())
+	if err != nil {
+		slog.Warn("Could not determine whether any profiles were stored on the keyring, defaulting to filesystem.", "error", err)
+	}
+	if hasKeyringStore {
+		slog.Debug("Keyring store still active, migrating profiles to filesystem.")
+		err := profiles.Migrate(profiles.ProfileDriverFileSystem, profiles.ProfileDriverKeyring)
+		if err != nil {
+			cli.ExitWithError(fmt.Sprintf("Error during profile migration from %s, to %s. Otdfctl cannot continue with profiles being stored within %s, please use the `profile migrate` command to manually migrate profiles", profiles.ProfileDriverKeyring, profiles.ProfileDriverFileSystem, profiles.ProfileDriverKeyring), err)
```

Comment:
We need to use the configured app name variable here for when otdfctl is consumed as a library.
```suggestion
			cli.ExitWithError(fmt.Sprintf("Error during profile migration from %s, to %s. %s cannot continue with profiles being stored within %s, please use the `profile migrate` command to manually migrate profiles", profiles.ProfileDriverKeyring, profiles.ProfileDriverFileSystem, config.AppName, profiles.ProfileDriverKeyring), err)
```